File: Leetcode/1323 - Maximum 69 Number/solution.js
var maximum69Number  = function(num) {
    let tempNum = num;
    let mul = 0;
    let currMul = 1;

    while (tempNum > 0) {
        if (tempNum % 10 === 6) {
            mul = currMul;
        }
        tempNum = Math.floor(tempNum / 10);
        currMul *= 10;
    }

    return num + (3 * mul);
};

File: Leetcode/1323 - Maximum 69 Number/solution.py
def maximum69Number (self, num: int) -> int:
    tempNum = num
    mul = 0
    currMul = 1

    while tempNum > 0:
        if tempNum % 10 == 6:
            mul = currMul
        tempNum //= 10
        currMul *= 10

    return num + (3 * mul)